src/ape_cache/query.py

```diff
@@ -80,61 +79,62 @@ def _block_query(
                     step=query.step,
                 )
                 return pd.DataFrame(columns=query.columns, data=q.fetchall())
+        raise Exception("SELECT statement improperly set.")
 
     @singledispatchmethod
     def table(self, query: QueryType):
         pass
 
     @table.register
-    def block_table(self, query: BlockQuery):
+    def block_table(self, query: BlockQuery) -> str:
         return "blocks"
 
     @table.register
-    def transactions_table(self, query: BlockTransactionQuery):
+    def transactions_table(self, query: BlockTransactionQuery) -> str:
         return "transactions"
 
     @table.register
-    def contract_events_table(self, query: ContractEventQuery):
+    def contract_events_table(self, query: ContractEventQuery) -> str:
         return "contract_events"
 
     @singledispatchmethod
-    def column(self, query: QueryType):
+    def column(self, query: QueryType) -> str:
         pass
 
     @column.register
-    def block_column(self, query: BlockQuery):
+    def block_column(self, query: BlockQuery) -> str:
         return "number"
 
     @column.register
-    def block_transaction_column(self, query: BlockTransactionQuery):
+    def block_transaction_column(self, query: BlockTransactionQuery) -> str:
         return "block_hash"
 
     @column.register
-    def contract_event_column(self, query: ContractEventQuery):
+    def contract_event_column(self, query: ContractEventQuery) -> str:
         return "block_number"
 
     @singledispatchmethod
-    def cache_query(self):
+    def cache_query(self) -> str:
         pass
```

Comment:
Don't think you'd need type ignores if this statement raised

Also, shouldn't this be `cache_query(self, query: QueryType)`?